Sources/Array+.swift

```diff
@@ -22,19 +22,33 @@ public extension Array where Element: Equatable {
     }
 }
 
-public extension Array where Element: Hashable {
-    public mutating func unify() {
-        self = unified()
-    }
-}
-
-public extension Collection where Element: Hashable {
-    public func unified() -> [Element] {
+public extension RangeReplacableCollection where Self: RandomAccessCollection, Element: Hashable {
+    /// Returns a collection containing the contents of the original collection with no duplicate elements.
+    ///
+    /// NOTE: The elements are not garanteed to be in the same order as the original colection.
+    ///
+    /// Complexity: O(n), where n is the number of elements in the collection.
+    public func unified() -> Self {
+        var set = Set<Element>()
+        var r = Self()
+        var newCount = 0
         return reduce(into: []) {
-            if !$0.contains($1) {
+            if set.insert($0).inserted {
                 $0.append($1)
+                newCount += 1
             }
         }
+        r.reserveCapacity(newCount)
```

Comment:
This line is unreachable because this function has already returned at line 35.

Reply:
Where does it return?

Reply:
This line:
https://github.com/tattn/SwiftExtensions/pull/7/files#diff-68771007cdfcaa7e091470870107adc0R35